tools/nuget/validate_package.py

```diff
@@ -57,13 +74,65 @@ def parse_arguments():
 def check_exists(path):
     return os.path.exists(path)
 
+def remove_residual_files(path):
+    if check_exists(path):
+        os.remove(path)
+
 
 def is_windows():
     return sys.platform.startswith("win")
 
 
+def check_if_headers_are_present(
+    header_files, header_folder, file_list_in_package, platform
+):
+    for header in header_files:
+        path = header_folder + "/" + header
+        print("Checking path: " + path)
+        if path not in file_list_in_package:
+            print(header + " not found for " + platform)
+            raise Exception(header + " not found for " + platform)
+
+def download_and_verify_directml_dlls(zip_file, platform, package_folder):
+    dml_nuget_spec_file_name = "Microsoft.ML.OnnxRuntime.DirectML.nuspec"
+    dml_nuget_package_name = 'Microsoft.AI.DirectML'
+    dml_nuget_package_version = '1.9.0'
```

Comment:
Shouldn't this be 1.9.1? I think this highlights one the problem of having to search and replace multiple instances of the DML version in this repo: there are other parts that reference 1.9.1, but the latest 1.12.1 release of ORT+DML is referencing DML 1.9.0.

Reply:
If the pipeline runs in a context where the repo source is available, then you might be better off parsing the DML package version from /packages.config.

Reply:
Yeah, I agree ideally the version upgrade should happen only at one place. I will add this as backlog item as this requires bit more involvement and right now there are other priority items which are due.

Reply:
ORT1.12.1 refers to DML 1.9.1. Thanks for pointing out. I will update the version to 1.9.1